test/test_client.py

```diff
@@ -0,0 +1,13 @@
+import unittest
+
+from phonon.client import ShardedClient
+
+
+class TestShardedClient(unittest.TestCase):
+
+    def setUp(self):
+        self.client = ShardedClient(hosts=['localhost1', 'localhost2'])
+
+    def test_route(self):
+        assert self.client.route('d') is self.client.clients[0]
+        assert self.client.route('b') is self.client.clients[1]
```

Comment:
Probably worth adding a couple more tests to verify that `__getattr__` is properly routing to a particular client if `args` are provided and otherwise getting written to all hosts.